test/test_exporter.py

```diff
@@ -327,127 +343,128 @@ def test_entire_batch_fail(self, mock_const, mock_post):
+        exporter = DynatraceMetricsExporter()
 
-        aggregator = aggregate.MinMaxSumCountAggregator()
-        aggregator.update(100)
-        aggregator.update(1)
-        self._update_value(aggregator, 10)
-        records.append(self._create_record(aggregator))
+        metric_reader = PeriodicExportingMetricReader(
+            export_interval_millis=3600000,  # 1h so that the test can finish before the collection event fires.
+            exporter=exporter)
 
-        aggregator = aggregate.LastValueAggregator()
-        self._update_value(aggregator, 20)
-        records.append(self._create_record(aggregator))
+        meter_provider = MeterProvider(metric_readers=[metric_reader])
+        meter = meter_provider.get_meter(name="my.meter", version="1.0.0")
+        counter = meter.create_counter("my.instr")
+        counter.add(10, attributes={"l1": "v1", "l2": "v2"})
 
-        exporter = DynatraceMetricsExporter()
-        exporter._is_delta_export = True
-        result = exporter.export(records)
+        metric_reader.collect()
 
-        expected = """my.instr,l1=v1,l2=v2,dt.metrics.source=opentelemetry count,delta=10
-my.instr,l1=v1,l2=v2,dt.metrics.source=opentelemetry gauge,min=1,max=100,sum=111,count=3
-my.instr,l1=v1,l2=v2,dt.metrics.source=opentelemetry gauge,20\n"""
-
-        self.assertEqual(MetricsExportResult.SUCCESS, result)
         mock_post.assert_called_once_with(
             self._ingest_endpoint,
-            data=expected,
+            data=AnyStringMatching(r"my\.instr,(l2=v2,l1=v1|l1=v1,l2=v2),dt\.metrics\.source=opentelemetry count,"
+                                   r"delta=10 [0-9]*\n"),
             headers=self._headers)
 
-    def _create_record(self, aggregator: aggregate.Aggregator):
-        return MetricRecord(
-            self._metric, self._labels, aggregator, Resource({})
-        )
-
-    @staticmethod
-    def _update_value(
-        aggregator: aggregate.Aggregator,
-        value: Union[int, float],
-    ):
-        aggregator.update(value)
-        aggregator.take_checkpoint()
-        # can be overwritten later
-        aggregator.last_update_timestamp = 0
+        # shut down cleanly to avoid failed exports later.
+        meter_provider.shutdown()
+
+    def _create_record(self, point: PointT):
+        return Metric(attributes=self._attributes,
+                      name=self._instrument_name,
+                      point=point,
+                      description="",
+                      unit="1",
+                      resource=Resource({}),
+                      instrumentation_info=InstrumentationInfo(name="dynatrace.opentelemetry.metrics.export",
+                                                               version="0.0.1"))
+
+    def _create_sum_data_point(self, value: int) -> Sum:
+        return Sum(start_time_unix_nano=self._test_timestamp_nanos,
+                   aggregation_temporality=AggregationTemporality.DELTA,
+                   is_monotonic=True,
```

Comment:
Do we also have tests with non-monotonic sums?

Reply:
Added in [e565f52](https://github.com/dynatrace-oss/opentelemetry-metric-python/pull/22/commits/e565f52f841c132dfbc9c483a70099425cd2edb7) :slightly_smiling_face: